### PR TITLE
Web page summary components

### PR DESCRIPTION
**Pull request by TonySchroeder**
close https://github.com/progwise/george-ai/issues/19

### CODE REVIEW

File: apps/playwright-scraper/src/strapi.ts [hunk 1]
```diff
@@ -3,132 +3,160 @@ import { WebPageSummary } from './main.js'
...
       })
+      console.log(
+        'Successfully created ScrapedWebPage with ID:',
+        response.createScrapedWebPage?.data?.id,
+      )
+      return
+    }
+
+    // Loop through WebPageSummary entries to find and update the matching AI model
+    const updatedWebPageSummaries =
```
Comment (moncapitaine) [inline apps/playwright-scraper/src/strapi.ts:117]: simpler

File: apps/playwright-scraper/src/strapi.ts [hunk 2]
```diff
@@ -3,132 +3,160 @@ import { WebPageSummary } from './main.js'
...
+        }
+        return entry
+      })
 
+    if (
+      updatedWebPageSummaries?.some(
+        (entry) => entry?.LargeLanguageModel === AI_MODEL,
+      )
+    ) {
+      await client.request(UPDATE_SCRAPE_WEBPAGE_MUTATION, {
```
Comment (moncapitaine) [inline apps/playwright-scraper/src/strapi.ts:133]: no second call needed. One call is enough. Make nice code to produce the right array, after that execute the mutation with this one and only array.

File: apps/playwright-scraper/src/strapi.ts [hunk 3]
```diff
@@ -3,132 +3,131 @@ import { WebPageSummary } from './main.js'
...
       }
     }
   }
 `)
 
-const GET_WEBPAGE_SUMMARY_BY_URL_AND_MODEL_QUERY = graphql(`
-  query GetWebPageSummary($url: String!, $model: String!) {
-    webPageSummaries(
+const GET_SCRAPE_WEBPAGE_BY_URL_QUERY = graphql(`
+  query GetScrapedWebPageByUrl($url: String!) {
```
Comment (PascalHelbig) [inline apps/playwright-scraper/src/strapi.ts:59]: Just a minor thing: To me the name suggests, that this query returns a single `ScrapedWebPage` but it is actually an array.

File: apps/playwright-scraper/src/strapi.ts [hunk 4]
```diff
@@ -3,132 +3,131 @@ import { WebPageSummary } from './main.js'
...
+          }
         }
       }
     }
   }
 `)
 
-export const upsertWebPageSummaries = async (summary: WebPageSummary) => {
+export const upsertScrapedWebPage = async (summary: WebPageSummary) => {
+  const inputData = {
```
Comment (PascalHelbig) [inline apps/playwright-scraper/src/strapi.ts:84]: Maybe `webPageSummary` is a better name?

File: apps/playwright-scraper/src/strapi.ts [hunk 5]
```diff
@@ -3,132 +3,131 @@ import { WebPageSummary } from './main.js'
...
       data {
         id
         attributes {
           Title
           Url
-          LargeLanguageModel
           OriginalContent
-          GeneratedSummary
-          GeneratedKeywords
+          WebPageSummary {
```
Comment (PascalHelbig) [inline apps/playwright-scraper/src/strapi.ts:28]: Why is it named `WebPageSummary` and not `WebPageSummaries`? Isn't it an array?

File: apps/playwright-scraper/src/strapi.ts [hunk 6]
```diff
@@ -3,132 +3,131 @@ import { WebPageSummary } from './main.js'
...
-
-    const existingSummaryResponse = await client.request(
-      GET_WEBPAGE_SUMMARY_BY_URL_AND_MODEL_QUERY,
-      {
-        url: summary.url,
-        model: 'gpt-3.5-turbo',
-      },
+    const summaries = existing?.attributes?.WebPageSummary
+    const updatedSummaries = summaries?.some(
+      (summary) => summary?.LargeLanguageModel === AI_MODEL,
```
Comment (PascalHelbig) [inline apps/playwright-scraper/src/strapi.ts:113]: I like the solution with a ternary operator.

However, I think it is more readable if we split it into two statements:
```tsx
const hasSummaryForLargeLanguageModel = summaries?.some(
  (summary) => summary?.LargeLanguageModel === AI_MODEL,
)
const updatedSummaries = hasSummaryForLargeLanguageModel
  ? ... 
  : ....
```

@TonySchroeder, what do you think?

File: apps/playwright-scraper/src/strapi.ts [hunk 7]
```diff
@@ -3,132 +3,131 @@ import { WebPageSummary } from './main.js'
...
-      Title: summary.title,
-      OriginalContent: summary.content,
-      GeneratedSummary: summary.summary,
-      GeneratedKeywords: JSON.stringify(summary.keywords),
-    }
+    const { scrapedWebPages } = await client.request(
+      GET_SCRAPE_WEBPAGE_BY_URL_QUERY,
+      { url: summary.url },
+    )
+    const existing = scrapedWebPages?.data?.[0]
```
Comment (PascalHelbig) [inline apps/playwright-scraper/src/strapi.ts:95]: Maybe `existingScrapedWebPage` is more clear?

**Review by moncapitaine — APPROVED** [review]

**Pull request merged**